### ISSUE TITLE
Possible false positive bug with destructured named exports

### ISSUE THREAD

**Issue by dburles** [jaydenseric/find-unused-exports#1]
Reproduction: https://github.com/dburles/find-unused-exports-issue

I found that exporting named exports in this way results in a false positive with no exports named:

```
index.js
  
1 unused export in 1 module.
```

**Comment by jaydenseric** [comment]
Interesting, maybe you could take a stab at a PR?

**Comment by jaydenseric** [comment]
Probably a destructured variable export hasn't been accounted for near here:

https://github.com/jaydenseric/find-unused-exports/blob/3208426379b6f70c7267a165e1aac7728fca2808/private/scanModuleCode.js#L111-L118

**Comment by jaydenseric** [comment]
Also, what about array destructuring? E.g. something like:

```js
export const [foo] = [true];
```

**Comment by jaydenseric** [comment]
Also, need to account for this:

```js
export const { a, b: c } = { a: true, b: true };
```

**Comment by jaydenseric** [comment]
What about this:

```js
export const [, a] = [true, true];
```

**Comment by jaydenseric** [comment]
Holy cow! Just released this needs to be a recursive operation as the destructuring can be arbitrarily deep/complex:

```js
export const { k, l: { m: [n]} } = { k: true, l: { m: [true] } };
```

In that case the named exports are `k` and `n`.

**Comment by jaydenseric** [comment]
Also, spread syntax:

```js
export const { a, ...b } = { a: true, b: true, c: true };
export const [, ...c] = [1, 2];
```

**Comment by jaydenseric** [comment]
The fix has been published in [v1.1.1](https://github.com/jaydenseric/find-unused-exports/releases/tag/v1.1.1) 🚀
